### PR TITLE
Add unfold()

### PR DESCRIPTION
**Pull request by pentacular**
This adds unfold() and diameter(), cleans up some output paths, and removes api/v1-pdf, v1-svg, v1-stl

### CODE REVIEW

**Comment by hrgdavor** [comment]
where can unfold be seen in action ? I am just curious :)

File: algorithm/cgal/Cast.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
+int Cast(Geometry* geometry) {
```
Comment (BarbourSmith) [inline algorithm/cgal/Cast.h:1]: Liking these new multiple files :)

**Review by BarbourSmith — APPROVED** [review]
Breaking the C++ stuff into multiple files was such a good idea. It's way easier to understand now I think

**Comment by BarbourSmith** [comment]
@hrgdavor It's still very much a work in progress I believe but the basic example is something like this from the PR:

image

Basically taking a 3D shape and "unfolding" into a 2D version, ideally the most compact 2D version with the minimum number of joints.

**Comment by pentacular** [comment]
> @hrgdavor It's still very much a work in progress I believe but the basic example is something like this from the PR:
> 
> image
> 
> Basically taking a 3D shape and "unfolding" into a 2D version, ideally the most compact 2D version with the minimum number of joints.

Yes, it produces a set of faces and fold edges, which can then be post-processed.

This is the kind of thing you could produce from those faces and edges for e.g., printing.

image

**Pull request merged**